### ISSUE TITLE
Large iCloud backup

### ISSUE THREAD

**Issue by GamerKingFaiz** [holzschu/a-shell#337]
The app is using 2.9 GB of my iCloud storage in my device backup.

Looking at the app storage, 900 MB is the app install and 2 GB is from documents & data.

I don't recall saving that much data. Is there a way to check what is taking up all that data?

Here's a screenshot of my iPad storage.
![Screenshot 2021-10-31 at 5 28 57 PM](https://user-images.githubusercontent.com/9065136/139606743-f7fcaafa-86d5-4baa-ac9a-0e8d6379e934.png)

**Comment by holzschu** [comment]
Yes. The terminal command, `du` will tell you how much space is used by a given folder. It's easier if you have an idea of where the files might be. If you've been using shortcuts a lot, for example with youtube-dl, they might be in `~shortcuts` or in `~/tmp` (but `~/tmp` is not supposed to be backed up on iCloud, I think). 

Short version:
```
du -skh ~/*
du -skh ~shortcuts/*
```

**Comment by GamerKingFaiz** [comment]
It seems like most of the space is in the `./.Trash` and `.` directories. Are these ashell directories or iPad directories?

If ashell how can I delete them?

If iPad why is ashell backing up local files?

![Screenshot 2021-11-01 at 9 39 33 AM](https://user-images.githubusercontent.com/9065136/139707809-a66b43a9-85be-44ab-a649-b30bd7a9dc72.png)

**Comment by holzschu** [comment]
The total for `.` is the total of everything inside `.`, including `./.Trash`, so basically everything is in `.Trash` (except 100K bytes in `youtube-upload-master`). Based on the rest of the info you provide, I think that is `~/Documents/.Trash` (3757360 blocks of 512 bytes (`du` without `-k` option) is equal to 1.8G).

If I remember correctly, this `.Trash` folder is where iOS stores deleted files for a while, so you can un-delete them if needed (like the garbage bin on MacOS and Windows). You will probably see those files in the "Recently Deleted" folder in the Files app, and you can force-delete them there (long press on the file, scroll to "Delete now"). You can also force-delete them from inside the app with `rm -rf ~/Documents/.Trash/*` (although you might want to check what's in there first with `ls ~/Documents/.Trash/`).

**Comment by GamerKingFaiz** [comment]
I force deleted my trash in my iPad library, but there was still some files in ashell. I then ran the `rm -rf ~/Documents/.Trash/*` command and that cleaned up everything. I turned iCloud back up off and then on again for ashell and it successfully reduced my backup size.

I'll keep an eye on this going forward, but I'm afraid it'll back up my trash again. I think it'd be ideal if backing up the trash could be disabled.

**Comment by holzschu** [comment]
I have no control over which files iOS choses to backup on iCloud.
